wagtail/admin/views/chooser.py

```diff
@@ -129,10 +130,54 @@ def browse(request, parent_page_id=None):
     parent_page.can_choose = can_choose_page(
         parent_page, permission_proxy, desired_classes, can_choose_root, user_perm, target_pages=target_pages, match_subclass=match_subclass)
 
+    selected_locale = None
+    locale_options = []
     show_locale_labels = getattr(settings, 'WAGTAIL_I18N_ENABLED', False)
     if show_locale_labels:
         pages = pages.select_related('locale')
 
+        # 'locale' is the current value of the "Locale" selector in the UI
```

Comment:
I think this part is only relevant when browsing the root page (same with the `if selected_locale` block below)

Reply:
It is indeed. Tweaked based on this and the comment below.

aside: adding the switcher at the root level was a last minute tweak, so looking at the whole thing with fresh eyes was useful